### ISSUE TITLE
Bug - React introduction: The article on Medium article comparing class and function components requires paid subscription for reading

### ISSUE THREAD

**Issue by twentysixhugs** [TheOdinProject/curriculum#23821]
Complete the following REQUIRED checkboxes:
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this issue follows the `Bug - location of bug: brief description of bug` format, e.g. `Bug - React section: Knowledge Checks don't link to resource`

The following checkbox is OPTIONAL:

-   [ ] I would like to be assigned this issue to work on it

<hr>

**1. Description of the Bug:**

- In the section components of the lesson, there's a link to a medium article comparing class and functional components.
Later, in the knowledge check, there's a question about the information from the article that links to it.
It requires paid subscription on medium to be read.
- [Link to the article](https://medium.com/@Zwenza/functional-vs-class-components-in-react-231e3fbd7108)
- Well, I can pretty much open it in private browsing mode and read, but it may confuse some learners, so I want to draw your attention to it.

**2. How To Reproduce:**
Open the article from a medium account that has no member-only subscriptions and has read all member-only free stories.

**3. Expected Behavior:**
Something else describing it :)

**Comment by wise-king-sullyman** [comment]
I agree, if a suitable replacement resource can be found moving away from medium would be nice. Thank you for putting up this issue!

**Comment by KonradLach** [comment]
I would like to be assigned to the issue!

**Comment by thatblindgeye** [comment]
@KonradLach been a while, but have you made any progress on this issue?
